cli/medperf/commands/view.py

```diff
@@ -0,0 +1,84 @@
+import yaml
+import json
+from typing import Union
+
+from medperf import config
+from medperf.entities.interface import Entity
+from medperf.exceptions import InvalidArgumentError
+
+
+class EntityView:
+    @staticmethod
+    def run(
+        entity_id: Union[int, str],
+        entity_class: Entity,
+        format: str = "yaml",
+        local_only: bool = False,
+        mine_only: bool = False,
+        output: str = None,
+    ):
+        """Displays the contents of a single or multiple entities of a given type
+
+        Args:
+            entity_id (Union[int, str]): Entity identifies
+            entity_class (Entity): Entity type
+            local_only (bool, optional): Display all local entities. Defaults to False.
+            mine_only (bool, optional): Display all current-user entities. Defaults to False.
+            format (str, optional): What format to use to display the contents. Valid formats: [yaml, json]. Defaults to yaml.
+            output (str, optional): Path to a file for storing the entity contents. If not provided, the contents are printed.
+        """
+        entity_view = EntityView(
+            entity_id, entity_class, format, local_only, mine_only, output
+        )
+        entity_view.validate()
+        entity_view.prepare()
+        if output is None:
+            entity_view.display()
+        else:
+            entity_view.store()
+
+    def __init__(self, entity_id, entity_class, format, local_only, mine_only, output):
+        self.entity_id = entity_id
+        self.entity_class = entity_class
+        self.format = format
+        self.local_only = local_only
+        self.mine_only = mine_only
+        self.output = output
+        self.data = []
+
+    def validate(self):
+        valid_formats = set(["yaml", "json"])
```

Comment:
just curious :) why set?

Reply:
Because checking for the existence of an element in a set is O(1), although in this specific scenario we don't really gain much given that the check happens only once 😅